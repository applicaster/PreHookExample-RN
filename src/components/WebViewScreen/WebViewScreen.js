import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  View,
  WebView,
} from 'react-native';
import BackButton from '../BackButton';

class WebViewScreen extends Component {
  static navigationOptions = ({ navigation, screenProps, params }) => ({
      ...screenProps,
      headerTitle: navigation.state.params.headerTitle,
      headerLeft: <BackButton onPress={() => {navigation.goBack(); }} />,
  });

  constructor(props) {
    super(props);
    this.state = { url: props.eventOriginUrl };
    this.onNavigationStateChange = this.onNavigationStateChange.bind(this);
  }

  onNavigationStateChange(webViewState) {
    const { url } = webViewState;
    const { eventOriginUrl } = this.props;
    if (url === 'https://www.instagram.com/') {
      this.setState({ url: `${eventOriginUrl}?r=${Date.now()}` });
    }
  }

  render() {
    const { url } = this.state;
    const backgroundFeedColor = { backgroundColor: this.context.backgroundColor };
  
    return (
      <WebView source={{uri: url }} style={[backgroundFeedColor]} onNavigationStateChange={this.onNavigationStateChange} />
    );
  }
}

WebViewScreen.propTypes = {
  eventOriginUrl: PropTypes.string,
};

WebViewScreen.contextTypes = {
  backgroundColor: PropTypes.string,
};

export default WebViewScreen;
