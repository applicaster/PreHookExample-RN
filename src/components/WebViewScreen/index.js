import { connect } from 'react-redux';
import {
  getActiveEventOriginUrl,
} from '../../selectors';

import WebViewScreen from './WebViewScreen';

const mapStateToProps = state => ({
  eventOriginUrl: getActiveEventOriginUrl(state),
});

export default connect(mapStateToProps)(WebViewScreen);
