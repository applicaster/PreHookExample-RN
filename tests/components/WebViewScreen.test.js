import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import WebViewScreen from '../../src/components/WebViewScreen/WebViewScreen';

jest.mock('WebView', () => 'WebView');

describe('WebViewScreen', () => {
  test('component renders correctly', () => {
    const tree = renderer.create(
      <WebViewScreen />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
