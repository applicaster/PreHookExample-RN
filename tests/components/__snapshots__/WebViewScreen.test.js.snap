// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebViewScreen component renders correctly 1`] = `
<WebView
  onNavigationStateChange={[Function]}
  source={
    Object {
      "uri": undefined,
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
/>
`;
